lib/ansible/utils/display.py

```diff
@@ -678,12 +720,11 @@ def prompt_until(self, msg, private=False, seconds=None, interrupt_input=None, c
 
     def _read_non_blocking_stdin(
         self,
-        echo=False,  # type: bool
-        seconds=None,  # type: int
-        interrupt_input=None,  # type: t.Iterable[bytes]
-        complete_input=None,  # type: t.Iterable[bytes]
-    ):  # type: (...) -> bytes
-
+        echo: bool = False,
+        seconds: int | None = None,
+        interrupt_input: t.Iterable[bytes] | None = None,
+        complete_input: t.Iterable[bytes] | None = None,
```

Comment:
There's the same problem with bytes being iterable here. I checked the function body and only saw `is (not) None` checks and container inclusion checks. So the actual type is
```suggestion
        interrupt_input: collections.abc.Container[bytes] | None = None,
        complete_input: collections.abc.Container[bytes] | None = None,
```

Reply:
But technically, I have a hunch that the invocations might only use bytes, and that would be
```suggestion
        interrupt_input: bytes | None = None,
        complete_input: bytes | None = None,
```
You choose.